### ISSUE TITLE
Reverse relation still exposes deleted objects

### ISSUE THREAD

**Issue by decibyte** [scoursen/django-softdelete#41]
I think I've found a bug, but it might be me doing something wrong.

Have a look at [the test case project](https://github.com/decibyte/django-softdelte-testcase) I created to show what the bug is.

I've created 2 simple models: `Parent` and `Child`. `Child` has a `OneToOneField` relation to `Parent`.

If I create one of each -- child relating to parent -- and then delete the child, then the child is still accessible via `parent.child`.

I would expect this to raise a `DoesNotExist`, as the child is (soft)deleted.

**Comment by natehawkboss** [comment]
Yes, I have the same issue.  It is actually causing some problems for me as it was pulling up deleted records in a form and filling it out, but then obviously not explicitly un-deleting the record when it was saved, just updating the deleted record.

**Comment by adiTrullion** [comment]
Hey, any idea how to make it work?

**Comment by rizkyarlin** [comment]
I am currently working on a similar case. For me it's for checking if the children exists in a ReverseManyToOne. Still working on a PoC.

But for OneToOneField, I believe it's a little bit trickier.


For OneToOneField  I'm thinking about doing something like:

overwriting the __init__() of django.db.models.base.Model

to do something like:
```python
            if is_related_object:
                # If we are passed a related instance, set it using the
                # field.name instead of field.attname (e.g. "user" instead of
                # "user_id") so that the object gets properly cached (and type
                # checked) by the RelatedObjectDescriptor.
                if rel_obj is not _DEFERRED:
                    if isinstance(rel_obj, SoftDeleteObject):
                        if rel_obj.deleted_at is not None:
                            raise ObjectDoesNotExist()
                    _setattr(self, field.name, rel_obj)
```
